### ISSUE TITLE
undefined reference to `avcodec_send_packet'

### ISSUE THREAD

**Issue by tosiara** [Motion-Project/motion#338]
Updated to the latest https://github.com/Motion-Project/motion/commit/b5c3a73cc62e7641e550cf742a90a74f78bdae33 and unable to link on one machine:
```
Linking Motion...
--------------------------------------------------------------------------------
gcc -L/usr/local/lib  -o motion motion.o logger.o conf.o draw.o jpegutils.o video_loopback.o video_v4l2.o video_common.o video_bktr.o netcam.o netcam_ftp.o netcam_jpeg.o netcam_wget.o track.o alg.o event.o picture.o rotate.o webhttpd.o stream.o md5.o netcam_rtsp.o ffmpeg.o  -lm -L/usr/local/lib -pthread -ljpeg   -pthread -L/home/odroid/ffmpeg/out/lib -lavformat -lavcodec -lXv -lXext -lvdpau -lva -lxcb -lX11 -ljack -lasound -lx264 -lpthread -llzma -lbz2 -lz -lswresample -lswscale -lavutil -lm  
netcam_rtsp.o: In function `rtsp_decode_video':
/home/odroid/motion/netcam_rtsp.c:87: undefined reference to `avcodec_send_packet'
/home/odroid/motion/netcam_rtsp.c:95: undefined reference to `avcodec_receive_frame'
netcam_rtsp.o: In function `netcam_open_codec':
/home/odroid/motion/netcam_rtsp.c:205: undefined reference to `avcodec_parameters_to_context'
ffmpeg.o: In function `my_avcodec_close':
/home/odroid/motion/ffmpeg.c:149: undefined reference to `avcodec_free_context'
ffmpeg.o: In function `ffmpeg_encode_video':
/home/odroid/motion/ffmpeg.c:334: undefined reference to `avcodec_send_frame'
/home/odroid/motion/ffmpeg.c:340: undefined reference to `avcodec_receive_packet'
ffmpeg.o: In function `my_avcodec_close':
/home/odroid/motion/ffmpeg.c:149: undefined reference to `avcodec_free_context'
ffmpeg.o: In function `ffmpeg_set_stream':
/home/odroid/motion/ffmpeg.c:604: undefined reference to `avcodec_parameters_from_context'
collect2: error: ld returned 1 exit status
make: *** [motion] Error 1
```
ARM Ubuntu 13, ffmpeg built from sources. Previous version (back few commits) compiles fine
Other ARM device updated and built successfully
What could be wrong?

**Comment by tosiara** [comment]
Narrowed down, the issues is caused exactly by this commit: https://github.com/Motion-Project/motion/commit/b5c3a73cc62e7641e550cf742a90a74f78bdae33

**Comment by Mr-Dave** [comment]
What are the precise version numbers for libavformat, libavcodec, etc.?

**Comment by Mr-DaveDev** [comment]
Can you also verify that for this build you start with a <code>make clean</code> and then the <code>make</code>

**Comment by tosiara** [comment]
I clone the repo from scratch. See full compile log and ffmpeg version at the bottom:
```
root@odroid:/home/odroid# git clone https://github.com/Motion-Project/motion.git
Cloning into 'motion'...
remote: Counting objects: 5260, done.
remote: Compressing objects: 100% (35/35), done.
remote: Total 5260 (delta 13), reused 0 (delta 0), pack-reused 5225
Receiving objects: 100% (5260/5260), 3.81 MiB | 817.00 KiB/s, done.
Resolving deltas: 100% (3954/3954), done.
Checking connectivity... done
root@odroid:/home/odroid# cd motion
root@odroid:/home/odroid/motion# autoreconf -fiv
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force 
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: running: /usr/bin/autoconf --force
autoreconf: running: /usr/bin/autoheader --force
autoreconf: configure.ac: not using Automake
autoreconf: Leaving directory `.'
root@odroid:/home/odroid/motion# ./configure --with-ffmpeg=/home/odroid/ffmpeg/out-3.1
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking for ANSI C header files... (cached) yes
checking for an ANSI C-conforming const... yes
checking for Darwin/BSD... no
checking linux/videodev2.h usability... yes
checking linux/videodev2.h presence... yes
checking for linux/videodev2.h... yes
checking pthread.h usability... yes
checking pthread.h presence... yes
checking for pthread.h... yes
checking for threads... yes
checking setjmp.h usability... yes
checking setjmp.h presence... yes
checking for setjmp.h... yes
checking jerror.h usability... yes
checking jerror.h presence... yes
checking for jerror.h... yes
checking jpeglib.h usability... yes
checking jpeglib.h presence... yes
checking for jpeglib.h... yes
checking jpg libraries... yes
checking for pkg-config... yes
checking sqlite3.h usability... no
checking sqlite3.h presence... no
checking for sqlite3.h... no
checking mysql/mysql.h usability... no
checking mysql/mysql.h presence... no
checking for mysql/mysql.h... no
checking for mysql support... no
checking postgresql/libpq-fe.h usability... no
checking postgresql/libpq-fe.h presence... no
checking for postgresql/libpq-fe.h... no
checking for PostgreSQL support... no
checking stdio.h usability... yes
checking stdio.h presence... yes
checking for stdio.h... yes
checking for unistd.h... (cached) yes
checking for stdint.h... (cached) yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking time.h usability... yes
checking time.h presence... yes
checking for time.h... yes
checking signal.h usability... yes
checking signal.h presence... yes
checking for signal.h... yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/mman.h usability... yes
checking sys/mman.h presence... yes
checking for sys/mman.h... yes
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking for sys/types.h... (cached) yes
configure: creating ./config.status
config.status: creating camera1-dist.conf
config.status: creating camera2-dist.conf
config.status: creating camera3-dist.conf
config.status: creating camera4-dist.conf
config.status: creating motion-dist.conf
config.status: creating motion.init-FreeBSD.sh
config.status: creating motion.init-Debian
config.status: creating motion.service
config.status: creating motion.spec
config.status: creating Makefile
config.status: creating config.h

   **************************
      Configure status       
      motion 4.0.1+giteb137c5
   **************************

OS             :     Linux
pthread support:     Yes
jpeg support:        Yes
webp support:        No
V4L2 support:        Yes
BKTR support:        No
MMAL support:        No
 ... libraspberrypi-dev package not installed
FFmpeg support:      Yes
 ... FFMPEG_CFLAGS: -I/home/odroid/ffmpeg/out-3.1/include  
 ... FFMPEG_LIBS: -pthread -L/home/odroid/ffmpeg/out-3.1/lib -lavdevice -lavfilter -lswscale -lpostproc -lavformat -lavcodec -lXv -lXext -lvdpau -lva -lxcb -lX11 -ljack -lasound -lSDL -lx264 -lpthread -llzma -lbz2 -lz -lswresample -lavutil -lm  
SQLite3 support:     No
MYSQL support:       No
PostgreSQL support:  No

CFLAGS: -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE 
LIBS: -lm -L/usr/local/lib -pthread -ljpeg
LDFLAGS:  -L/usr/local/lib 

Install prefix:       /usr/local

root@odroid:/home/odroid/motion# make
Welcome to the setup procedure for Motion, the motion detection daemon! If you get
error messages during this procedure, please report them to the mailing list. The
Motion Guide contains all information you should need to get Motion up and running.

Version: 4.0.1+giteb137c5
Platform: Linux (if this is incorrect, please read README.FreeBSD)

Generating dependencies, please wait...

Compiling Motion object files...
--------------------------------------------------------------------------------
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o motion.o motion.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o logger.o logger.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o conf.o conf.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o draw.o draw.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o jpegutils.o jpegutils.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o video_loopback.o video_loopback.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o video_v4l2.o video_v4l2.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o video_common.o video_common.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o video_bktr.o video_bktr.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o netcam.o netcam.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o netcam_ftp.o netcam_ftp.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o netcam_jpeg.o netcam_jpeg.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o netcam_wget.o netcam_wget.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o track.o track.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o alg.o alg.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o event.o event.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o picture.o picture.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o rotate.o rotate.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o webhttpd.o webhttpd.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o stream.o stream.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o md5.o md5.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o netcam_rtsp.o netcam_rtsp.c
gcc -g -O2 -I/usr/local/include -g -O2 -D_THREAD_SAFE  -Wall -DVERSION=\"4.0.1+giteb137c5\" -Dsysconfdir=\"/usr/local/etc\" -I/home/odroid/ffmpeg/out-3.1/include      -c -o ffmpeg.o ffmpeg.c
--------------------------------------------------------------------------------
Motion object files compiled.

Linking Motion...
--------------------------------------------------------------------------------
gcc -L/usr/local/lib  -o motion motion.o logger.o conf.o draw.o jpegutils.o video_loopback.o video_v4l2.o video_common.o video_bktr.o netcam.o netcam_ftp.o netcam_jpeg.o netcam_wget.o track.o alg.o event.o picture.o rotate.o webhttpd.o stream.o md5.o netcam_rtsp.o ffmpeg.o  -lm -L/usr/local/lib -pthread -ljpeg   -pthread -L/home/odroid/ffmpeg/out-3.1/lib -lavdevice -lavfilter -lswscale -lpostproc -lavformat -lavcodec -lXv -lXext -lvdpau -lva -lxcb -lX11 -ljack -lasound -lSDL -lx264 -lpthread -llzma -lbz2 -lz -lswresample -lavutil -lm  
netcam_rtsp.o: In function `rtsp_decode_video':
/home/odroid/motion/netcam_rtsp.c:87: undefined reference to `avcodec_send_packet'
/home/odroid/motion/netcam_rtsp.c:95: undefined reference to `avcodec_receive_frame'
netcam_rtsp.o: In function `netcam_open_codec':
/home/odroid/motion/netcam_rtsp.c:205: undefined reference to `avcodec_parameters_to_context'
ffmpeg.o: In function `my_avcodec_close':
/home/odroid/motion/ffmpeg.c:149: undefined reference to `avcodec_free_context'
ffmpeg.o: In function `ffmpeg_encode_video':
/home/odroid/motion/ffmpeg.c:334: undefined reference to `avcodec_send_frame'
/home/odroid/motion/ffmpeg.c:340: undefined reference to `avcodec_receive_packet'
ffmpeg.o: In function `my_avcodec_close':
/home/odroid/motion/ffmpeg.c:149: undefined reference to `avcodec_free_context'
ffmpeg.o: In function `ffmpeg_set_stream':
/home/odroid/motion/ffmpeg.c:603: undefined reference to `avcodec_parameters_from_context'
collect2: error: ld returned 1 exit status
make: *** [motion] Error 1

root@odroid:/home/odroid/ffmpeg/out-3.1# ./bin/ffmpeg 
ffmpeg version n3.1.7-3-ge1ed229 Copyright (c) 2000-2017 the FFmpeg developers
  built with gcc 4.8 (Ubuntu/Linaro 4.8.1-10ubuntu9)
  configuration: --prefix=/home/odroid/ffmpeg/out-3.1 --enable-libx264 --enable-gpl
  libavutil      55. 28.100 / 55. 28.100
  libavcodec     57. 48.101 / 57. 48.101
  libavformat    57. 41.100 / 57. 41.100
  libavdevice    57.  0.101 / 57.  0.101
  libavfilter     6. 47.100 /  6. 47.100
  libswscale      4.  1.100 /  4.  1.100
  libswresample   2.  1.100 /  2.  1.100
  libpostproc    54.  0.100 / 54.  0.100
```

**Comment by Mr-Dave** [comment]
The referenced functions are included within ffmpeg 3.1.  I would recommend rebuilding the ffmpeg and then validating that the libavformat.pc and other package configuration files are all pointing to the correct locations.

**Comment by tosiara** [comment]
Pulled latest https://github.com/Motion-Project/motion/commit/eb137c5f5dffc0613a2459a39b855122918a71dc - issue resolved
